cmd/oras/internal/option/remote.go

```diff
@@ -243,6 +251,21 @@ func (opts *Remote) authClient(registry string, debug bool) (client *auth.Client
 	return
 }
 
+func (opts *Remote) parseCustomHeaders() error {
+	if len(opts.headerFlags) != 0 {
+		headers := map[string][]string{}
+		for _, h := range opts.headerFlags {
+			name, csv, found := strings.Cut(h, ":")
+			if !found || strings.TrimSpace(csv) == "" {
+				return fmt.Errorf("cannot parse headers: %q", h)
+			}
+			headers[name] = append(headers[name], strings.Split(csv, ",")...)
```

Comment:
We should consume the value as it is without splitting.

You can try `-H "foo: a" -H "foo: b,c"` with `curl`. It will generate
```
> foo: a
> foo: b,c
```

Reply:
changed according to Billy's comment above.